packages/backend/src/open_payments/receiver/model.ts

```diff
@@ -39,45 +67,31 @@ export class Receiver extends ConnectionBase {
       : undefined
     const receivedAmount = parseAmount(incomingPayment.receivedAmount)
 
-    return this.fromOpenPaymentsConnection(
-      incomingPayment.ilpStreamConnection,
-      incomingAmount?.value,
-      receivedAmount.value,
-      expiresAt
-    )
-  }
-
-  private static fromOpenPaymentsConnection(
-    connection: OpenPaymentsConnection,
-    incomingAmountValue?: bigint,
-    receivedAmountValue?: bigint,
-    expiresAt?: Date
-  ): Receiver | undefined {
-    const ilpAddress = connection.ilpAddress
-
-    if (!isValidIlpAddress(ilpAddress)) {
+    if (!isValidIlpAddress(incomingPayment.ilpStreamConnection.ilpAddress)) {
       return undefined
     }
 
     return new this(
       {
-        id: connection.id,
-        assetCode: connection.assetCode,
-        assetScale: connection.assetScale,
-        sharedSecret: connection.sharedSecret,
-        ilpAddress
+        ...incomingPayment.ilpStreamConnection,
+        ilpAddress: incomingPayment.ilpStreamConnection.ilpAddress
       },
-      incomingAmountValue,
-      receivedAmountValue,
-      expiresAt
+      {
+        id: incomingPayment.id,
+        completed: incomingPayment.completed,
+        paymentPointer: incomingPayment.paymentPointer,
+        createdAt: new Date(incomingPayment.createdAt),
+        updatedAt: new Date(incomingPayment.updatedAt),
+        expiresAt,
+        receivedAmount,
+        incomingAmount
+      }
     )
   }
 
   private constructor(
     connection: OpenPaymentsConnectionWithIlpAddress,
-    private readonly incomingAmountValue?: bigint,
-    private readonly receivedAmountValue?: bigint,
-    public readonly expiresAt?: Date
+    public incomingPayment?: ReceiverIncomingPayment
```

Comment:
I decided to place the full `incomingPayment` on the Receiver object. This makes sure we don't lose all of the information when creating a receiver from a payment & and as a result, allows me to keep the `CreateReceiverResponse` GraphQL type filled with the same payment information as before (`paymentPointer` `description` `expiresAt` etc)

I think ideally, I would prefer to have `type Receiver = Connection | IncomingPayment` and then just use typeguards to determine what kind of Receiver it is without needing a class. We would lose the logic in the `Receiver.fromIncomingPayment` static method, but maybe we could convert those to a few simple functions if we need to use them.

Reply:
`type Receiver = Connection | IncomingPayment` also sounds appealing to me.
Maybe some of the `fromIncomingPayment` could be moved to the `isReceiverIncomingPayment` type guard.

Reply:
Since `receiverService` will require some changes (in order to replace some of those methods it has to call `remoteIncomingPayment` service) I could make an attempt using `type Receiver = Connection | IncomingPayment` instead.